File: go/x402/signers/evm/okx_signer.go
package evm

import (
	"errors"

	x402evm "github.com/okx/payments/go/x402/mechanisms/evm"
)

// TEEConfig holds configuration for TEE-based signing (Phase 2).
type TEEConfig struct {
	Endpoint  string
	AccessKey string
}

// OKXSignerConfig configures the OKX signer.
// Provide either PrivateKey (Phase 1) or TEE (Phase 2).
type OKXSignerConfig struct {
	// PrivateKey is a hex-encoded ECDSA private key for local EIP-3009 signing.
	PrivateKey string

	// TEE configures TEE-based signing. Not yet implemented (Phase 2).
	TEE *TEEConfig
}

// NewOKXSigner creates a ClientEvmSigner based on the provided config.
// Phase 1: only private key signing is supported.
// Phase 2: TEE signing will be added with automatic fallback.
func NewOKXSigner(config OKXSignerConfig) (x402evm.ClientEvmSigner, error) {
	if config.TEE == nil && config.PrivateKey == "" {
		return nil, errors.New("must provide either tee or privateKey")
	}

	if config.TEE != nil {
		return nil, errors.New("TEE signing not yet implemented (Phase 2)")
	}

	return NewClientSignerFromPrivateKey(config.PrivateKey)
}

File: python/x402/src/x402/mechanisms/evm/okx_signer.py
"""OKX signer factory for creating EVM signers from OKX configuration."""

from __future__ import annotations

from dataclasses import dataclass

from .signer import ClientEvmSigner


@dataclass
class TEEConfig:
    """Configuration for TEE-based signing (not yet implemented)."""

    endpoint: str
    access_key: str


@dataclass
class OKXSignerConfig:
    """Configuration for OKX EVM signer.

    Provide either private_key or tee.
    """

    private_key: str | None = None
    tee: TEEConfig | None = None


def new_okx_signer(config: OKXSignerConfig) -> ClientEvmSigner:
    """Create a ClientEvmSigner from OKX configuration.

    Currently only private key signing is supported.
    TEE signing is not yet implemented.

    Args:
        config: Signer configuration with either private_key or tee.

    Returns:
        A ClientEvmSigner implementation.

    Raises:
        ValueError: If neither private_key nor tee is provided.
        NotImplementedError: If tee is provided.
    """
    if config.tee is None and not config.private_key:
        raise ValueError("must provide either tee or private_key")

    if config.tee is not None:
        raise NotImplementedError("TEE signing is not yet implemented")

    from eth_account import Account

    from .signers import EthAccountSigner

    key = config.private_key
    if key and not key.startswith("0x"):
        key = "0x" + key

    account = Account.from_key(key)
    return EthAccountSigner(account)
